provider/adapters/src/any_payload.rs

```diff
@@ -47,6 +40,34 @@ pub struct AnyPayloadProvider {
     pub data: AnyPayload,
 }
 
+impl AnyPayloadProvider {
+    /// Creates an `AnyPayloadProvider` with an owned (allocated) payload of the given data.
+    pub fn new_owned<M: ResourceMarker + 'static>(data: M::Yokeable) -> Self {
+        AnyPayloadProvider {
+            key: M::KEY,
+            data: AnyPayload::from_rc_payload::<M>(alloc::rc::Rc::from(DataPayload::from_owned(
```

Comment:
Thought: It's not _great_ to hide an rc instantiation inside of a function, but it's a lot of boilerplate and this is not a very hot function so it's fine.